src/__tests__/Autocomplete.test.tsx

```diff
@@ -354,29 +359,29 @@ describe('Autocomplete', () => {
       // current sort order matches the result of `sortOnCloseFnMock`
       expect(sortOnCloseFnMock).toHaveBeenCalledTimes(mockItems.length - 1)
       if (inputNode) {
-        userEvent.type(inputNode, 'ze')
+        await user.type(inputNode, 'ze')
         // eslint-disable-next-line github/no-blur
         fireEvent.blur(inputNode)
       }
 
       // wait a tick for blur to finish
-      setTimeout(() => {
-        expect(sortOnCloseFnMock).toHaveBeenCalledTimes(mockItems.length)
-      }, 0)
+      await waitFor(() => expect(sortOnCloseFnMock).toHaveBeenCalled())
```

Comment:
Here fixing the `setTimeout` actually exposed an assertion failure that was previously hidden. `Array.sort` does not call the sort function once per element like `filter` or `map` would; a sorting algorithm has to run more than one single pass over the array to fully sort it. In this case, it was calling the sort function 18 times on an array of 10 elements. Rather than test the number of calls, which depends on the JS engine's `sort` implementation, I changed it to just test that at least one call was made.